src/Agent.Listener/Configuration/NativeWindowsServiceHelper.cs

```diff
@@ -415,12 +416,91 @@ public void InstallService(string serviceName, string serviceDisplayName, string
                     processInvoker.ExecuteAsync(string.Empty, agentServiceExecutable, "init", null, default(System.Threading.CancellationToken)).GetAwaiter().GetResult();
                 }
 
+                //set recovery option to restart on failure.
+                ArrayList failureActions = new ArrayList();
+                //first failure
+                failureActions.Add(new FailureAction(RecoverAction.Restart, 0));
+                //second failure
+                failureActions.Add(new FailureAction(RecoverAction.Restart, 60000));
+                //subsequent failures
+                failureActions.Add(new FailureAction(RecoverAction.Restart, 60000));
+
+                // Err check var
+                bool result = false;
+
+                // Lock the Service Database
+                svcLock = LockServiceDatabase(scmHndl);
+                if (svcLock.ToInt64() <= 0)
+                {
+                    throw new Exception("Failed to Lock Service Database for Write");
+                }
+
+                int[] actions = new int[failureActions.Count * 2];
+                int currInd = 0;
+                foreach (FailureAction fa in failureActions)
+                {
+                    actions[currInd] = (int)fa.Type;
+                    actions[++currInd] = fa.Delay;
+                    currInd++;
+                }
+
+                // Need to pack 8 bytes per struct
+                tmpBuf = Marshal.AllocHGlobal(failureActions.Count * 8);
+                // Move array into marshallable pointer
+                Marshal.Copy(actions, 0, tmpBuf, failureActions.Count * 2);
+
+                // Set the SERVICE_FAILURE_ACTIONS struct
+                SERVICE_FAILURE_ACTIONS sfa = new SERVICE_FAILURE_ACTIONS();
+                sfa.cActions = failureActions.Count;
+                sfa.dwResetPeriod = SERVICE_NO_CHANGE;
+                sfa.lpCommand = String.Empty;
+                sfa.lpRebootMsg = String.Empty;
+                sfa.lpsaActions = tmpBuf.ToInt64();
+
+                // Call the ChangeServiceFailureActions() abstraction of ChangeServiceConfig2()
+                result = ChangeServiceFailureActions(svcHndl, SERVICE_CONFIG_FAILURE_ACTIONS, ref sfa);
+
+                //Check the return
+                if (!result)
+                {
+                    if (GetLastError() == ReturnCode.ERROR_ACCESS_DENIED)
+                    {
+                        throw new Exception("Access Denied while setting Failure Actions");
```

Comment:
Per offline discussion, a summary of changes is:
- Print "Configuring Windows Service recovery options."
- If fails for any reason, throw so we exit non-zero (configuration did not complete cleanly).